src/Lib.hs

```diff
@@ -1,56 +1,97 @@
-{-# LANGUAGE DataKinds          #-}
-{-# LANGUAGE DerivingStrategies #-}
-{-# LANGUAGE TypeOperators      #-}
+{-# LANGUAGE DataKinds           #-}
+{-# LANGUAGE DerivingStrategies  #-}
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeOperators       #-}
 
 module Lib
   ( User(..)
   , myServer
   , myApi
   ) where
 
-import           Data.Aeson               as JSON
-import           Data.Foldable            (find)
-import           Models                   (User (..))
+import           Control.Monad.IO.Class     (MonadIO (liftIO))
+import           Data.Functor               (void)
+import           Data.Pool                  (Pool, withResource)
+import           Database.PostgreSQL.Simple (Connection, Only (Only), execute,
+                                             query, query_)
+import           Models                     (User (..))
 import           Servant
 
 type UsersAPI
   = "users" :> "list-all" :> Get '[ JSON] [User] 
       :<|> "users" :> Capture "usrId" Int :> Get '[ JSON] User 
-      :<|> "createUser" :> ReqBody '[ JSON] User :> Post '[ JSON] User 
-      :<|> "deleteUser" :> Capture "usrId" Int :> Delete '[ JSON] [User] 
-      :<|> "updateUser" :> Capture "usrId" Int :> ReqBody '[ JSON] User :> Put '[ JSON] [User]
+      :<|> "deleteUser" :> Capture "usrId" Int :> Delete '[ JSON] NoContent 
+      :<|> "upsetUser" :> ReqBody '[ JSON] User :> Put '[ JSON] NoContent
 
-myUsers :: [User]
-myUsers = [User 0 "charly" $ Just 25, User 1 "gon" $ Just 22]
+getUserById :: Pool Connection -> Int -> IO (Maybe User)
+getUserById conns uid =
+  handleDbResponse
+    <$> withResource
+          conns
+          (\conn ->
+             query
+               conn
+               "select * from users where id = ?"
+               (Only uid :: Only Int))
+  where
+    handleDbResponse userResponse =
+      case userResponse of
+        []    -> Nothing
+        (x:_) -> Just x
 
--- TODO: move to utils
-getUserById :: Int -> Maybe User
-getUserById usrId = find (\user -> userId user == usrId) myUsers
+deleteUserById :: Pool Connection -> Int -> IO NoContent
+deleteUserById conns uid = do
+  _ <-
+    withResource
+      conns
+      (\conn ->
+         execute conn "delete from users where id = ?" (Only uid :: Only Int))
+  return NoContent
```

Comment:
It's fine that the return type of the delete function is empty, but `NoContent` is a Servant-specific type. Since this function is just a DB function, it would be better to return something like `()`, which signals the same behavior to the caller but decouples it from Servant. The handler should take care of interpreting this `()` value as a `NoContent` response.